src/nss-resolve/nss-resolve.c

```diff
@@ -463,11 +465,22 @@ enum nss_status _nss_resolve_gethostbyname3_r(
         }
 
         const char *canonical = p.name ?: name;
+        size_t canonical_len = strlen(canonical);
 
-        size_t alen = FAMILY_ADDRESS_SIZE(af);
-        size_t l = strlen(canonical);
+        r = dns_name_normalize(name, /* flags = */ 0, &alias);
+        if (r < 0)
+                goto fail;
+        r = dns_name_equal(canonical, alias);
+        if (r < 0)
+                goto fail;
+        size_t alias_len = (r == 0 ? strlen(alias) : 0);
```

Comment:
the outer () are unnecessary

Reply:
i'd probably also free `alias` here if the identical to `canonical`, and then check for `alias` in the rest of the code to decide whether we should care or not.